PINCache/PINDiskCache.m

```diff
@@ -771,14 +778,21 @@ - (NSURL *)fileURLForKey:(NSString *)key updateFileModificationDate:(BOOL)update
     }
     
     NSDate *now = [[NSDate alloc] init];
-    NSURL *fileURL = nil;
+    NSURL *fileURL = [self encodedFileURLForKey:key];
     
     [self lock];
-        fileURL = [self _locked_encodedFileURLForKey:key];
-        
         if ([[NSFileManager defaultManager] fileExistsAtPath:[fileURL path]]) {
             if (updateFileModificationDate) {
-                [self _locked_setFileModificationDate:now forURL:fileURL];
+                //It's unlikely these need to be run in order or to be run quickly. Get it done eventually instead of blocking.
+                __weak PINDiskCache *weakSelf = self;
+                dispatch_async(_asyncQueue, ^{
```

Comment:
Seems worth factoring out this sub-block into a method, maybe including the dispatch_async?